src/format_bed.py

```diff
@@ -8,6 +8,16 @@
     parse_line, print_line
 )
 
+def parser(inputFile):
+    outLines = []
+    lineBegin, lineEnd = 0, 0
+    for i, c in enumerate(inputFile):
+        if c == "\n":
+            lineEnd = i
+            outLines.append(parse_line(inputFile[lineBegin:lineEnd]))
+            lineBegin = i + 1
+    return outLines
```

Comment:
This is a curious function, and I am not sure it works. It depends a bit on what is in `inputFile`, but if it is the lines in a file, then you only have `c == "\n"` for empty lines, and you don't have many of those in the input. Also, unless `lineBegin == lineEnd - 1`, the slice you do will return a list, and `parse_line()` cannot handle lists but only strings.

I don't see you calling the function anywhere, so it doesn't matter that it doesn't work, but you probably don't want it. You certainly don't need it.